src/libraries/System.Data.Common/src/System/Data/SQLTypes/SQLGuid.cs

```
@@ -328,6 +333,12 @@ public static XmlQualifiedName GetXsdType(XmlSchemaSet schemaSet)
             return new XmlQualifiedName("string", XmlSchema.Namespace);
         }
 
+        void ISerializable.GetObjectData(SerializationInfo si, StreamingContext sc)
+        {
+            ArgumentNullException.ThrowIfNull(si);
```

Check is unnecessary; just let the NRE propagate out.